### PR TITLE
Check if the provided user has access to the individual projects 

### PR DESCRIPTION
**Pull request by nakhbari**
Use octokit client API for github to check if the repository exists. If the repository is private and the specified user does not have access, it will return false.

TESTED_LOCALLY

### CODE REVIEW

File: services/config_service.rb [hunk 1]
```diff
@@ -54,18 +54,9 @@ def octokit_projects(provider_credential: nil)
...
-      # proper paging, or we ask for specific repos instead
-      # Either way, my account has access to too many repos, so for now, let's just workaround using this
-      current_repo_git_url_set << "https://github.com/nakhbari/Hello"
-      current_repo_git_url_set << "https://github.com/taquitos/ci-sample-repo"
-      current_repo_git_url_set << "https://github.com/fastlane/ci"
-
       logger.debug("Finding projects we have access to with #{provider_credential.ci_user.email}, #{provider_credential.type}")
       projects = self.project_service.projects.select do |project|
-        current_repo_git_url_set.include?(project.repo_config.git_url)
+        current_code_hosting_service.access_to_repo?({repo: project.repo_config.git_url})
```
Comment (taquitos) [inline services/config_service.rb]: `access_to_repo?({repo: project.repo_config.git_url})` can be:
`access_to_repo?(repo: project.repo_config.git_url)`

File: services/code_hosting/code_hosting_service.rb [hunk 2]
```diff
@@ -18,5 +18,9 @@ def provider_type
     def repos
       not_implemented(__method__)
     end
+
+    def access_to_repo?(repo: nil)
```
Comment (taquitos) [inline services/code_hosting/code_hosting_service.rb]: in this case, I think we actually want it be maybe
`access_to_repo?(repo_config: nil)`
or even 
`access_to_repo?(repo_url: nil)`
The reason being, we have a `GitRepo` object already and that is very heavyweight.

File: services/code_hosting/git_hub_service.rb [hunk 3]
```diff
@@ -41,6 +41,12 @@ def repos
       client.repos({}, query: { sort: "asc" })
     end
 
+    # Does the client with the associated credentials have access to the specified repo?
+    # @repo [String] Repo URL as string
+    def access_to_repo?(repo: nil)
+      client.repository?(repo.sub("https://github.com/",""))
```
Comment (taquitos) [inline services/code_hosting/git_hub_service.rb]: yeah, in this case, it's not a `repo` it's just a string. You could either rename it `git_url` or pass in a `repo_config` item and extract the `git_url` from there.

Comment (nakhbari) [reply services/code_hosting/git_hub_service.rb]: if i name the parent `repo_url` i'll still have to use that name correct? Can the child change the name of the param?

Comment (taquitos) [reply services/code_hosting/git_hub_service.rb]: you'll need to use `repo_url` if you change it to that. The child cannot change the name of a named param.

File: services/code_hosting/code_hosting_service.rb (hunk 2 above)
Comment (nakhbari) [reply services/code_hosting/code_hosting_service.rb]: changed to repo_url, to be repo type agnostic.

File: services/config_service.rb (hunk 1 above)
Comment (nakhbari) [reply services/config_service.rb]: done

File: services/code_hosting/git_hub_service.rb (hunk 3 above)
Comment (nakhbari) [reply services/code_hosting/git_hub_service.rb]: done

**Pull request merged**